votor/src/alpenglow_metrics.rs

```diff
@@ -0,0 +1,184 @@
+use {
+    histogram::Histogram,
+    solana_clock::{Epoch, Slot},
+    solana_pubkey::Pubkey,
+    solana_votor_messages::vote::Vote,
+    std::{
+        collections::BTreeMap,
+        time::{Duration, Instant},
+    },
+};
+
+/// Returns a [`Histogram`] configured for the use cases for this module.
+///
+/// Keeps the default precision and reduces the max value to 10s to get finer grained resolution.
+fn build_histogram() -> Histogram {
+    Histogram::configure()
+        .max_value(10_000_000)
+        .build()
+        .unwrap()
+}
+
+/// Tracks all [`Vote`] metrics for a given node.
+#[derive(Debug)]
+struct NodeVoteMetrics {
+    notar: Histogram,
+    notar_fallback: Histogram,
+    skip: Histogram,
+    skip_fallback: Histogram,
+    final_: Histogram,
+}
+
+impl Default for NodeVoteMetrics {
+    fn default() -> Self {
+        let histogram = build_histogram();
+        Self {
+            notar: histogram.clone(),
+            notar_fallback: histogram.clone(),
+            skip: histogram.clone(),
+            skip_fallback: histogram.clone(),
+            final_: histogram,
+        }
+    }
+}
+
+impl NodeVoteMetrics {
+    /// Records metrics for when `vote` was received after `elapsed` time has passed since the start of the slot.
+    fn record_vote(&mut self, vote: &Vote, elapsed: Duration) {
+        let elapsed = elapsed.as_micros();
+        let elapsed = match elapsed.try_into() {
+            Ok(e) => e,
+            Err(err) => {
+                warn!(
+                    "recording duration {elapsed} for vote {vote:?}: conversion to u64 failed \
+                     with {err}"
+                );
+                return;
+            }
+        };
+        let res = match vote {
+            Vote::Notarize(_) => self.notar.increment(elapsed),
+            Vote::NotarizeFallback(_) => self.notar_fallback.increment(elapsed),
+            Vote::Skip(_) => self.skip.increment(elapsed),
+            Vote::SkipFallback(_) => self.skip_fallback.increment(elapsed),
+            Vote::Finalize(_) => self.final_.increment(elapsed),
+        };
+        match res {
+            Ok(()) => (),
+            Err(err) => {
+                warn!(
+                    "recording duration {elapsed} for vote {vote:?}: recording failed with {err}"
+                );
+            }
+        }
+    }
+}
+
+/// Errors returned from [`AgMetrics::record_vote`].
+#[derive(Debug)]
+pub enum RecordVoteError {
+    /// Could not find start of slot entry.
+    SlotNotFound,
+}
+
+/// Errors returned from [`AgMetrics::record_block_hash_seen`].
+#[derive(Debug)]
+pub enum RecordBlockHashError {
+    /// Could not find start of slot entry.
+    SlotNotFound,
+}
+
+/// Tracks various Alpenglow related metrics.
+pub struct AgMetrics {
```

Comment:
very minor nit, would name this `Metrics` or `ConsensusMetrics` (same with the file). Essentially want to avoid calling anything that's already in the `votor` crate Ag - and one day in the future we won't have TowerBFT so everything will be alpenglow anyway.

Reply:
makes sense.  Renamed.